### ISSUE TITLE
add MediaType.isCompatible method

### ISSUE THREAD

**Issue by ghost** [google/guava#1903]
A method that compares just the `type` and `subtype` and ignores parameters. `MediaType.is()` could then delegate to this new method. 

Same functionality as `javax.ws.rs.core.MediaType`:

http://docs.oracle.com/javaee/7/api/javax/ws/rs/core/MediaType.html#isCompatible(javax.ws.rs.core.MediaType)

This comparison can be done using the current api by calling `withoutParameters()` on both types, but this may unnecessarily allocate a new `MediaType` on each invocation.

**Comment by abhinav-1504** [comment]
Hi, I’d like to work on this issue if it’s still relevant. I checked the current `MediaType` implementation and this feature still seems useful for parameter-insensitive compatibility checks without allocating intermediate instances via `withoutParameters()`.

Before preparing a PR, I wanted to confirm whether the maintainers would still consider adding such a method today.

**Comment by cpovirk** [comment]
I don't think we know enough about what we'd want here. It's possible that the best API is the one that we offer now, `a.withoutParameters().equals(b.withoutParameters())`, because it's explicit in a way that "`isCompatible`" is not. (I'd imagine that, in a fair number of cases, the caller can skip one of the two `withoutParameters` calls because one of the values is known not to have parameters.) It would be possible to optimize `withoutParameters` if the performance is a concern. Even that is borderline but might be worthwhile if it turns out to be straightforward.
